### PR TITLE
PR: Add missing `QtGui` utility function to `QtCore.Qt` for PySide bindings

### LINKED ISSUE #311: `QtCore.Qt.mightBeRichText` undefined in PySide
it seems we have to define for Pyside6
` QtCore.Qt.mightBeRichText = QtGui.Qt.mightBeRichText `

image

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #311 

Edit: Also fixes an issue with PyQt6 and sip to use `mightBeRichText`:

imagen

### CODE REVIEW

File: qtpy/QtCore.py [hunk 1]
```diff
@@ -58,6 +67,11 @@
     import PySide6.QtCore
     __version__ = PySide6.QtCore.__version__
 
+    # Missing QtGui utility functions on Qt
+    if getattr(Qt, 'mightBeRichText', None) is None:
+        from PySide6.QtGui import Qt as guiQt
```
Comment (ccordoba12) [inline qtpy/QtCore.py:72]: I think `guiQt` should be removed after the assignment in the line just before this one.

File: qtpy/QtCore.py [hunk 2]
```diff
@@ -79,5 +93,14 @@
 
     import PySide2.QtCore
     __version__ = PySide2.QtCore.__version__
+
+    # Missing QtGui utility functions on Qt
+    if getattr(Qt, 'mightBeRichText', None) is None:
+        try:
+            from PySide2.QtGui import Qt as guiQt
```
Comment (ccordoba12) [inline qtpy/QtCore.py:100]: The same here.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @dalthviz for this! I left two small comments, otherwise looks good to me.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,7 +17,7 @@ jobs:
   test:
     name: Test ${{ matrix.os }} Python ${{ matrix.python-version }} conda=${{ matrix.use-conda }}
     runs-on: ${{ matrix.os }}
-    timeout-minutes: 10
+    timeout-minutes: 12
```
Comment (CAM-Gerlach) [inline .github/workflows/ci.yml:20]: I was on my trip when you merged this earlier, but something like `15` minutes might be safer—particularly with the macOS builds, I've seen it sometimes stall for 5 or perhaps even 10 minutes on one step and then recover and succeed, mostly driven (as far as I can tell) by slowness and non-determinism in the conda solver.

File: qtpy/QtCore.py [hunk 4]
```diff
@@ -24,6 +24,15 @@
     from PyQt6.QtCore import QDateTime
     QDateTime.toPython = QDateTime.toPyDateTime
 
+    # For issue #311
+    # Seems like there is an error with sip. Without first
+    # trying to import `PyQt6.QtGui.Qt`
+    # Some functions like `PyQt6.QtCore.Qt.mightBeRichText` are missing
```
Comment (CAM-Gerlach) [inline qtpy/QtCore.py:30]:
```suggestion
    # Seems like there is an error with sip. Without first
    # trying to import `PyQt6.QtGui.Qt`, some functions like
    # `PyQt6.QtCore.Qt.mightBeRichText` are missing.
```

Fix punctuation/caps

**Review by CAM-Gerlach — APPROVED** [review]
Looks like @ccordoba12 's suggestions were implemented, so this is ready to go with a couple trivial comments

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @dalthviz!

**Review by CAM-Gerlach — APPROVED** [review]
LGTM now, thanks @dalthviz !

**Comment by CAM-Gerlach** [comment]
Merging so we can fix the tests in the other PRs.

File: qtpy/QtCore.py [hunk 5]
```diff
@@ -24,6 +24,15 @@
     from PyQt6.QtCore import QDateTime
     QDateTime.toPython = QDateTime.toPyDateTime
 
+    # For issue #311
+    # Seems like there is an error with sip. Without first
+    # trying to import `PyQt6.QtGui.Qt`, some functions like
+    # `PyQt6.QtCore.Qt.mightBeRichText` are missing.
+    try:
+        from PyQt6.QtGui import Qt
```
Comment (StSav012) [inline qtpy/QtCore.py:32]: Just out of curiosity, shouldn't this overwrite the `Qt` imported from `PyQt6.QtCore` a bit earlier?

**Review by StSav012 — COMMENTED** [review]
I've stumbled upon the issue #311 in the comments. Although it's merged ages ago, I'd like to discuss the fix a little.

What versions of `PyQt6`, `PySide2`, and `PySide6` require the fix? I can't tell whether the code is long obsolete or still needed. Please clarify this somewhere.

**Comment by dalthviz** [comment]
> What versions of PyQt6, PySide2, and PySide6 require the fix? I can't tell whether the code is long obsolete or still needed. Please clarify this somewhere.

Due the dates the issue was raised and fixed I would guess that is needed for Qt 6.1 or 6.2 bindings (although not completely sure :thinking:). Also, did a quick check using PySide2 5.15.2.1 and I got this:

image

So the fix is at least required by PySide 5.15.2.1. Could you give it a try to the Qt 6.2 bindings to check if the fix is still necessary there?

**Comment by StSav012** [comment]
So, the tests I've conducted gave the following results:

Flavor       | `QtCore.qVersion()` | `hasattr(QtCore.Qt, "mightBeRichText")` | `hasattr(QtGui.Qt, "mightBeRichText"))`
-----------|----------------------|---------------------------------------------|-------------------------------------------
`PySide2` | `5.13.2`, `5.15.2` | `False` | `True`
`PySide6` | `6.2.4`, `6.5.3`, `6.6.0` | `False` | `True`
`PyQt5`    | `5.13.2`, `5.15.2` | `True` | `'PyQt5.QtGui' has no attribute 'Qt'`
`PyQt6`    | `6.2.4`, `6.5.3`, `6.6.0` | `True` | `'PyQt6.QtGui' has no attribute 'Qt'`

So, the fix _is_ required for all current `PySide` versions, but shouldn't be here for `PyQt6`.

But that's not all.

Flavor       | `set(dir(QtGui.Qt)) - set(dir(QtCore.Qt))`
-----------|------------------------
`PySide2` | `{'convertFromPlainText', 'codecForHtml', 'mightBeRichText'}`
`PySide6` | `{'convertFromPlainText', 'mightBeRichText'}`

Shouldn't `convertFromPlainText` be defined for  `PyQt5` and `PyQt6` then, too?

```python
def convertFromPlainText(plain: str, mode: Qt.WhiteSpaceMode = Qt.WhiteSpaceMode.WhiteSpacePre) -> str:
    import html

    plain = html.escape(plain)
    if mode == Qt.WhiteSpaceMode.WhiteSpacePre:
        plain = plain.replace(" ", "\xa0").replace("\t", "\xa0" * 4)
    return "<p>" + plain + "</p>"
```

**Comment by dalthviz** [comment]
Thanks for the new info @StSav012 ! Could you open a new issue with your findings? Seems like we need to work in a couple of things indeed

**Comment by StSav012** [comment]
I'm unsure whether it's worth it. I've run `[a for a in dir(QtCore.Qt) if a[0].islower()]` (to filter the constants and classes out) for various flavors.

| Flavor | `[a for a in dir(QtCore.Qt) if a[0].islower()]` |
|--------|--------|
| `PyQt5-Qt5==5.15.2` | `['black', 'blue', 'color0', 'color1', 'cyan', 'darkBlue', 'darkCyan', 'darkGray', 'darkGreen', 'darkMagenta', 'darkRed', 'darkYellow', 'gray', 'green', 'lightGray', 'magenta', 'red', 'transparent', 'white', 'yellow']` |
| `PySide2==5.13.2` | `['black', 'blue', 'color0', 'color1', 'cyan', 'darkBlue', 'darkCyan', 'darkGray', 'darkGreen', 'darkMagenta', 'darkRed', 'darkYellow', 'gray', 'green', 'lightGray', 'magenta', 'red', 'transparent', 'white', 'yellow']` |
| `PyQt6-Qt6==6.6.0` | `['bin', 'bom', 'center', 'dec', 'endl', 'fixed', 'flush', 'forcepoint', 'forcesign', 'hex', 'left', 'lowercasebase', 'lowercasedigits', 'noforcepoint', 'noforcesign', 'noshowbase', 'oct', 'reset', 'right', 'scientific', 'showbase', 'uppercasebase', 'uppercasedigits', 'ws']` |
| `PySide6-Essentials==6.6.0` | `['beginPropertyUpdateGroup', 'bin', 'bom', 'center', 'dec', 'endPropertyUpdateGroup', 'endl', 'fixed', 'flush', 'forcepoint', 'forcesign', 'hex', 'left', 'lowercasebase', 'lowercasedigits', 'noforcepoint', 'noforcesign', 'noshowbase', 'oct', 'reset', 'right', 'scientific', 'showbase', 'uppercasebase', 'uppercasedigits', 'ws']` |

All but the two first ones differ. Does anyone need the functions to spend time porting them?

As for `convertFromPlainText`, it's nowhere to be found. Weirdly. Not even in `PyQt*.QtCore`. So, the flavors _are_ compatible in this sense.

**Comment by dalthviz** [comment]
> All but the two first ones differ. Does anyone need the functions to spend time porting them?

I guess for the moment no since no issue has been opened regarding that but it's good to know the variations that we could potentially handle

And thanks for opening the issue!

**Pull request merged**